include/keyword_vpd_parser.hpp

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include "parser_interface.hpp"
+#include "types.hpp"
+
+namespace vpd
+{
+
+/**
+ * @brief Concrete class to implement Keyword VPD parsing
+ *
+ * The class inherits ParserInterface class and overrides the parser
+ * functionality to implement parsing logic for Keyword VPD format.
+ */
+class KeywordVpdParser : public ParserInterface
+{
+  public:
+    KeywordVpdParser() = delete;
+    KeywordVpdParser(const KeywordVpdParser&) = delete;
+    KeywordVpdParser(KeywordVpdParser&&) = delete;
+    ~KeywordVpdParser() = default;
+
+    /**
+     * @brief Constructor
+     *
+     * Move kwVpdVector to parser object's m_keywordVpdVector
+     */
+    KeywordVpdParser(const types::BinaryVector& kwVpdVector) :
+        m_keywordVpdVector(kwVpdVector)
+    {
+    }
+
+    /**
+     * @brief A wrapper function to parse the keyword VPD binary data.
+     *
+     * It validates certain tags and checksum data, calls helper function
+     * to parse and emplace the data as keyword-value pairs in KeywordVpdMap.
+     *
+     * @error/exception DataException - VPD is not valid
+     * @return map of keyword:value
+     */
+    std::variant<types::ParsedVPD, types::KeywordVpdMap> parse();
+
+  private:
+    /**
+     * @brief Parse the vpd data and emplace as pair into the Map.
+     *
+     * @error/exception DataException - VPD data size is 0, check vpd
+     * @return map of keyword:value
+     */
+    types::KeywordVpdMap parser();
```

Comment:
Please rename the api. parser under parse api is not making much sense.
something like "populateVPDMap".

Reply:
sure